### ISSUE TITLE
setting a property does not update in Notion

### ISSUE THREAD

**Issue by benjixx** [mayneyao/notabase#7]
I have created a simple Notion database where I am trying to update a text property.

I am the owner of the workspace, I am using the token, I can read from the actual Notion database via notabase.

But when I try to update a single property of a row as described in https://github.com/mayneyao/notabase/tree/ts#write I don't get any error and within the Node process the property is updated (reading from it gives the new value).

But for some reason the update is not propagated to Notion.
Any ideas what might be wrong?

**Comment by mayneyao** [comment]
I just tested it and it worked for me. Do you have write permission?

**Comment by benjixx** [comment]
Very strange. Yes, I have write permission. I created a fresh Notion workspace for that test, created a Notion database within that. Afterwards I read the token from the browser, and copied the complete database URL as well.

Could you show me the working code?

**Comment by benjixx** [comment]
It is working now for me. I was trying to run it in a serverless environment which most likely had some restriction / didn't wait for the data to be written to Notion.
I now tried it locally again via a simply typescript file and run it via ts-node. That worked for me. So, all good. Thanks for looking into it so quickly anyway.

**Comment by mayneyao** [comment]
> It is working now for me. I was trying to run it in a serverless environment which most likely had some restriction / didn't wait for the data to be written to Notion.
> I now tried it locally again via a simply typescript file and run it via ts-node. That worked for me. So, all good. Thanks for looking into it so quickly anyway.

cloudflare worker?

**Comment by benjixx** [comment]
No, I tried on Zeit NOW. Looking at the notabase code again it looks like this line https://github.com/mayneyao/notabase/blob/master/src/collection.js#L412 is not awaiting the Promise and as such the serverless process would exit as there's nothing to wait for.

As a workaround I appended the following to my serverless async request handler after updating a property on a Notion database via notabase:

```
await new Promise((r) => setTimeout(r, 2000));
```

This sleeps for 2 seconds as thus gives notabase time to fulfill the Promise in the background. It is a very dirty workaround but most likely completely fine for my simple use case.
But I wanted to mention it anyway as this could be something to be improved.

Thanks for this nice library. Among the other ones I had a quick glance so far, notabase is really simple to use.
